chia/wallet/nft_wallet/uncurry_nft.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+
+from chia.types.blockchain_format.program import Program
+from chia.wallet.puzzles.load_clvm import load_clvm
+
+SINGLETON_TOP_LAYER_MOD = load_clvm("singleton_top_layer_v1_1.clvm")
+NFT_MOD = load_clvm("nft_innerpuz.clvm")
+
+
+@dataclass
+class UncurriedNFT:
+    """
+    A simple solution for uncurry NFT puzzle.
+    Initial the class with a full NFT puzzle, it will do a deep uncurry.
+    This is the only place you need to change after modified the Chialisp curried parameters.
+    """
+
+    matched: bool = field(init=False)
```

Comment:
`init=False` is not used much.  An instance of the class should mostly be able to be constructed by a single constructor call instead of a totally generic constructor call and then 20 attribute assignments.  In some cases, once you reach that structure you can make the class instances frozen/immutable.

Reply:
Note that this ends up leaving the class with maybe-not-even-present attributes.

```python-console
Python 3.8.10 (default, Mar 15 2022, 12:22:08) 
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import dataclasses
>>> @dataclasses.dataclass
... class C:
...     x: int = dataclasses.field(init=False)
... 
>>> c = C()
>>> c
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.8/dataclasses.py", line 368, in wrapper
    result = user_function(self)
  File "<string>", line 3, in __repr__
AttributeError: 'C' object has no attribute 'x'
```